src/main/webapp/app/overview/course-lectures/video-unit/video-unit.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { SafeResourceUrl } from '@angular/platform-browser';
+import { VideoUnit } from 'app/entities/lecture-unit/videoUnit.model';
+import { SafeResourceUrlPipe } from 'app/shared/pipes/safe-resource-url.pipe';
+
+@Component({
+    selector: 'jhi-video-unit',
+    templateUrl: './video-unit.component.html',
+    styleUrls: ['../../course-exercises/course-exercise-row.scss'],
+})
+export class VideoUnitComponent implements OnInit {
+    @Input()
+    videoUnit: VideoUnit;
+
+    youTubeUrl: SafeResourceUrl;
+
+    isCollapsed = true;
+
+    constructor(private safeResourceUrlPipe: SafeResourceUrlPipe) {}
+
+    ngOnInit() {
+        this.youTubeUrl = this.safeResourceUrlPipe.transform(`https://www.youtube.com/embed/${this.videoUnit.source}`);
```

Comment:
with this approach, we cannot support any other streaming platform. I don't think we should do that. I think it would be better to allow users to enter the whole URL and store it in the database instead. 
We could still have a special handling for Youtube URLs here, but we should not have a vendor lock-in!

Reply:
Ok I will switch it to storing the whole URL in the database instead of just the id.

Reply:
done in e92995fe232f9a7fd224e15317c68f6e7dad8f97